test/Tests/src/Semantic/Conversion_Spec.enso

```diff
@@ -202,6 +212,19 @@ spec =
             to_6 (v : Number & Decimal & Integer) = v
             to_6 m . should_equal 1.5
 
+        Test.specify "Requesting Integer & Fool" <|
+            do_stuff (x : Integer & Fool) =
```

Comment:
Copy of [test provided](https://github.com/enso-org/enso/pull/7769#pullrequestreview-1619920748) by @radeusgd with slight modifications using `x:Integer` to _hint the system_.

Implementing `==` for _multi value_ is was tricky - it is a method on `Any` and thus each value supports it. But at least the `==` and `to_text` work fine.